### PR TITLE
Add mechanism to simulate bulk sync delay

### PR DESCRIPTION
**Pull request by Tim-Brooks**
Currently Elasticsearch is used across a multitude of environments - local nvme ssds, spinning hd, nfs, and more. Most of our testing is targeting environments that delivery fast syncs. This commit adds a mechanism to induce sync delays for testing purposes.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/action/bulk/WriteAckDelayIT.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    fail(message);
+                }
+            } catch (Exception e) {
+                logger.error("search for all docs types failed", e);
+                if (firstError == null) {
+                    firstError = e;
+                }
+            }
+            try {
+                logger.debug("running search with a specific type");
```
Comment (henningandersen) [inline server/src/internalClusterTest/java/org/elasticsearch/action/bulk/WriteAckDelayIT.java:64]: I am sure I am missing something obvious... But the two searches look identical and do not seem to care about any "type"?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/internalClusterTest/java/org/elasticsearch/action/bulk/WriteAckDelayIT.java (hunk 1 above)
Comment (tbrooks8) [reply server/src/internalClusterTest/java/org/elasticsearch/action/bulk/WriteAckDelayIT.java:64]: Lol. I copied this from another test and probably kept around irrelevant code. I will adjust.

**Review by tlrx — COMMENTED** [review]
LGTM 2

**Pull request merged**